cmd/credentialUtil.go

```diff
@@ -537,66 +348,168 @@ func logAuthType(ct common.CredentialType, location common.Location, isSource bo
 
 var authMessagesAlreadyLogged = &sync.Map{}
 
+// isPublic reports true if the Blob URL passed can be read without auth.
+func isPublic(ctx context.Context, blobResourceURL string, cpkOptions common.CpkOptions) (isPublicResource bool) {
```

Comment:
should we consider adding unit tests for these standalone methods? 

isPublic, mdAccountNeedsOAuth, doGetCredentialTypeForLocation